src/browser/api/window.js

```diff
@@ -1502,8 +1502,28 @@ Window.moveTo = function(identity, left, top) {
 };
 
 Window.navigate = function(identity, url) {
-    let browserWindow = getElectronBrowserWindow(identity, 'navigate');
-    browserWindow.webContents.loadURL(url);
+    return new Promise((resolve, reject) => {
+        let browserWindow = getElectronBrowserWindow(identity, 'navigate');
+
+        let handleLoadFinished = success => {
+            browserWindow.webContents.removeListener('did-fail-load', didFail);
+            browserWindow.webContents.removeListener('did-finish-load', didSucceed);
+            if (success) {
+                resolve();
+            } else {
+                reject();
+            }
+        };
+
+        let didFail = () => handleLoadFinished(false);
+        let didSucceed = () => handleLoadFinished(true);
+        browserWindow.webContents.on('did-fail-load', didFail);
```

Comment:
Do we get the actual reason here (HTTP response code etc)? If so, lets not squash the reason but report it in the context from which it came.